src/ape/contracts/base.py

```diff
@@ -271,6 +276,13 @@ def __getattr__(self, attr_name: str) -> Any:
             any: The return value from the contract call, or a transaction receipt.
         """
 
+        if not self.is_contract:
```

Comment:
Note: Making this raise in `__getattr__` also causes the error to occur if just doing `MyContract.method` (without calling it). That may not be ideal. But I am struggling to find a way to get this to work with the call/transaction handler classes.

We could add this to the provider level but then we have to add it to basically all the providers.

Reply:
I mean on the actual `Contract*Handler.__call__(*args, **kwargs)` method have it raise if it is not a contract.

https://github.com/ApeWorX/ape/blob/main/src/ape/contracts/base.py#L114
https://github.com/ApeWorX/ape/blob/main/src/ape/contracts/base.py#L187

Reply:
Like, the interface is defined just fine, but when you make the call (e.g. `my_contract.myMethod(sender=a)`) have it raise there

Also consider adding a `kwarg` named `skip_contract_check=False` to those methods, so you can override this behavior for... reasons

Reply:
`ContractCallHandler` and `ContractTransactionHandler` do have a reference to the `AddressAPI` version.

Maybe I can refactor so that they do.